### ISSUE TITLE
Badge number does not get increased

### ISSUE THREAD

**Issue by nikomikulicic** [OneSignal/OneSignal-Unity-SDK#151]
**Description:**
Badge number does not get increased and is always 1. Exactly like this issue here:
https://github.com/OneSignal/OneSignal-iOS-SDK/issues/392 but with UnitySDK, instead of iOS. 

✅ Added the notification service extension, and set up an app group for app (group.bundleId.onesignal).
✅When sending notification through dashboard, I've selected "increase by 1" option. Notification is successfully received on device.
❌When sending push notification with image through dashboard, I do not receive image on device. As pointed out in issue linked above, the issue is probably in NotificationService not properly being set up, but I don't know what exactly could cause the issue.
✅NotificationService deployment target is 10.0

Issue linked above points out what could be the issue, but I don't know where to look next since NotificationService is set up by Unity and deployment target is properly set to 10.0. 

Possible important thing to note: I also have my own post build script which runs after OneSignal's and creates entitlements file. I had to comment out part of OneSignal's post build script where entitlements file is created. I didn't touch the part where NotificationService is added. I also made sure I added app groups, push notification and background modes (remote notifications) capabilities through my post build script.

**Environment**
OneSignal SDK 2.7.6.
Unity 2017.3.1f1
iOS 11.2.6.
Xcode 9.4.1.

**Comment by Nightsd01** [comment]
@nikomikulicic If you remove your own post build script, do badges/images work properly?

It's _definitely_ a problem with your notification service extension if both badges and images aren't working correctly.

**Comment by Nightsd01** [comment]
Also, in the project Unity generates, can you verify that the OneSignalNotificationServiceExtension is being embedded properly in your main app?

![binary](https://user-images.githubusercontent.com/8186991/46033751-c618e780-c0b3-11e8-8807-809da414ab2f.jpg)

**Comment by nikomikulicic** [comment]
@Nightsd01 , thank you for the response. I can confirm that OneSignalNotificationServiceExtension is embedded properly in main app.

Regarding removing my build script, I currently can't find an easy way to do it as there are other dependencies my project relies upon. Also, we are currently in a pre-release mode so I don't have time to try things out. I'll get back to you in couple of days if that's is ok. Thanks!

**Comment by Nightsd01** [comment]
@nikomikulicic Any updates?

**Comment by nikomikulicic** [comment]
@Nightsd01 , I'm sorry for the delay. 

I've just tried removing all dependencies I've had. Removed Fabric and Crashlytics (which also had post build scripts). Removed my own post build script. I'm currently left with unchanged OneSignal's post build script only.

Unfortunately, badges/images still don't work properly.

Here is screenshot of embedded frameworks in main target
![screen shot 2018-10-17 at 11 43 45](https://user-images.githubusercontent.com/19579227/47077915-5f926f80-d202-11e8-9ac0-bd124da7063c.png)

Main target's build phases:
![screen shot 2018-10-17 at 11 45 03](https://user-images.githubusercontent.com/19579227/47077962-789b2080-d202-11e8-8f3b-7c90721f561a.png)

Notification extension's frameworks:
![screen shot 2018-10-17 at 11 43 57](https://user-images.githubusercontent.com/19579227/47078015-9a94a300-d202-11e8-83bf-5e1a17d49c1f.png)
